### PR TITLE
Change task and executor name to the framework name

### PR DESCRIPTION
**Pull request by nicgrayson**
I want to be able to run multiple exhibitor clusters on the same mesos while using mesos-dns / mesos-consul.

The dns names for the exhibitor servers is generated from the executor name, i.e. exhibitor-0.service.consul. This means that I need different executor names for different clusters.

I was thinking about adding a task name or task prefix value but figured this solution was good since you want to know which task goes with which framework.

As you can see I ~~added the val frameworkName to Scheduler so I could access it from Exhibitor.scala then I~~ swapped out the hard coded 'exhibitor-' value. I also needed to change the logic of pulling the exhibitor server number out of this string since a framework name could include a dash.

Let me know if you have any questions or concerns.

This is almost exactly the same as the change in https://github.com/mesos/kafka/pull/206 btw

### CODE REVIEW

**Comment by serejja** [comment]
I think this only applies when you are using mesos-consul, as DNS names generated by mesos-dns include the framework name, like `$taskName.$frameworkName.mesos`.

Anyway, I think this is a good idea because even when you are looking at Mesos UI it's not always clear which framework tasks belong to.

The only thing I'm not quite clear with is why did you add the frameworkName field to the Scheduler. As I see, it could be just taken from Config, right? So that we wouldn't need to duplicate the frameworkName value in both Config and Scheduler.

Please let me know what you think. Once we're clear with this point I'm ready to merge your PR.

Thanks @nicgrayson!

**Comment by nicgrayson** [comment]
@serejja I was thinking I didn't have access to that class for some reason. Fixed.

File: src/main/scala/net/elodina/mesos/exhibitor/Exhibitor.scala [hunk 1]
```diff
@@ -46,7 +46,7 @@ case class Exhibitor(id: String) {
   def createTask(offer: Offer): TaskInfo = {
     val port = getPort(offer).getOrElse(throw new IllegalStateException("No suitable port"))
 
-    val name = s"exhibitor-${this.id}"
+    val name = s"${Scheduler.frameworkName}-${this.id}"
```
Comment (serejja) [inline src/main/scala/net/elodina/mesos/exhibitor/Exhibitor.scala]: oops? :)

Comment (nicgrayson) [inline src/main/scala/net/elodina/mesos/exhibitor/Exhibitor.scala]: Fixed already 😸

**Comment by nicgrayson** [comment]
@serejja rebased this to clean up the history 😄

**Pull request merged**